### PR TITLE
KHADRI-23102024-0644-PMMobile_Modify_Module

### PR DESCRIPTION
**Pull request by SalukulaMahendra**

### CODE REVIEW

File: src/main/java/com/khadri/jakartha/servlet/dao/ModifyDao.java
Comment (khadri2021) [inline src/main/java/com/khadri/jakartha/servlet/dao/ModifyDao.java:1]: package name typo mistake from jakartha to jakarta.

File: src/main/java/com/khadri/jakartha/servlet/modify/ModifyServlet.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+	@Override
+	protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+
+		String mobileModelNo = req.getParameter("MobileModelNumber");
+		String mobileName = req.getParameter("MobileName");
+		String mobileBrand = req.getParameter("MobileBrand");
+		String mobileCost = req.getParameter("MobileCost");
+		String mobileQty = req.getParameter("MobileQuantity");
+
+		ModifyForm form = new ModifyForm();
```
Comment (khadri2021) [inline src/main/java/com/khadri/jakartha/servlet/modify/ModifyServlet.java]: create form object at init method level.

File: src/main/java/com/khadri/jakartha/servlet/modify/ModifyServlet.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+		form.setMobileModelNumber(mobileModelNo);
+		form.setMobileName(mobileName);
+		form.setMobileBrand(mobileBrand);
+		form.setMobileCost(Double.parseDouble(mobileCost));
+		form.setMobileQuantity(Integer.parseInt(mobileQty));
+		int updateMovie = 0;
+		try {
```
Comment (khadri2021) [inline src/main/java/com/khadri/jakartha/servlet/modify/ModifyServlet.java]: what does exception occurs here?

File: src/main/java/com/khadri/jakartha/servlet/modify/ModifyServlet.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+	@Override
+	public void init() throws ServletException {
+		dao = new ModifyDao();
+	}
+
+	@Override
+	protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+
+		String mobileModelNo = req.getParameter("MobileModelNumber");
```
Comment (khadri2021) [inline src/main/java/com/khadri/jakartha/servlet/modify/ModifyServlet.java]: change from titled case to all lower case.

File: src/main/java/com/khadri/jakartha/servlet/modify/form/ModifyForm.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
+package com.khadri.jakartha.servlet.modify.form;
+
+public class ModifyForm {
+	private String mobileModelNumber;
+	private String mobileName;
+	private String mobileBrand;
+	private double mobileCost;
+	private int mobileQuantity;
+
+	public ModifyForm(String mobileModelNumber, String mobileName, String mobileBrand, double mobileCost,
```
Comment (khadri2021) [inline src/main/java/com/khadri/jakartha/servlet/modify/form/ModifyForm.java]: where you using args constructor?

File: src/main/webapp/WEB-INF/web.xml [hunk 5]
```diff
@@ -1,12 +1,27 @@
...
+	<display-name>workshope-servlet-mobile-shop-app</display-name>
+	<welcome-file-list>
+		<welcome-file>index.html</welcome-file>
+		<welcome-file>index.jsp</welcome-file>
+		<welcome-file>index.htm</welcome-file>
+		<welcome-file>default.html</welcome-file>
+		<welcome-file>default.jsp</welcome-file>
+		<welcome-file>default.htm</welcome-file>
+	</welcome-file-list>
+
```
Comment (khadri2021) [inline src/main/webapp/WEB-INF/web.xml]: why space here required.

File: src/main/webapp/WEB-INF/web.xml [hunk 6]
```diff
@@ -1,12 +1,27 @@
...
+
+	<servlet>
+		<servlet-name>modify</servlet-name>
+		<servlet-class>com.khadri.jakartha.servlet.modify.ModifyServlet</servlet-class>
+	</servlet>
+	<servlet-mapping>
+		<servlet-name>modify</servlet-name>
+		<url-pattern>/modify</url-pattern>
+
```
Comment (khadri2021) [inline src/main/webapp/WEB-INF/web.xml]: remove line space here.

File: src/main/webapp/index.html [hunk 7]
```diff
@@ -0,0 +1,16 @@
+<!DOCTYPE html>
```
Comment (khadri2021) [inline src/main/webapp/index.html:1]: remove line spaces.

File: src/main/java/com/khadri/jakartha/servlet/dao/ModifyDao.java [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+
+public class ModifyDao {
+	public int updateMovie(ModifyForm form) {
+		System.out.println("MobileDao updateMobile(-)");
+		int result = 0;
+		try {
+			Class.forName("com.mysql.cj.jdbc.Driver");
+
+			Connection con = DriverManager.getConnection(
+					"jdbc:MySQL://localhost:3306/2024_batch_servlet_mobile_shop_app", "root", "Wellcome@123");
```
Comment (khadri2021) [inline src/main/java/com/khadri/jakartha/servlet/dao/ModifyDao.java]: configure them as context parameters.

**Review by khadri2021 — COMMENTED** [review]
Develop the modification page as dynamic web page not html page and auto populate data for the corresponding model number.

**Review by thasmiyas39 — DISMISSED** [review]

File: src/main/java/com/khadri/jakartha/servlet/dao/ModifyDao.java
Comment (SalukulaMahendra) [reply src/main/java/com/khadri/jakartha/servlet/dao/ModifyDao.java:1]: change  package

File: src/main/java/com/khadri/jakartha/servlet/modify/ModifyServlet.java [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+
+import com.khadri.jakartha.servlet.dao.ModifyDao;
+import com.khadri.jakartha.servlet.modify.form.ModifyForm;
+
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServlet;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+
+public class ModifyServlet extends HttpServlet {
```
Comment (khadri2021) [inline src/main/java/com/khadri/jakartha/servlet/modify/ModifyServlet.java]: Change ModifyServlet to ModifyMobileServlet.

File: src/main/java/com/khadri/jakartha/servlet/modify/form/ModifyForm.java [hunk 10]
```diff
@@ -0,0 +1,63 @@
+package com.khadri.jakartha.servlet.modify.form;
+
+public class ModifyForm {
```
Comment (khadri2021) [inline src/main/java/com/khadri/jakartha/servlet/modify/form/ModifyForm.java]: Maintain same form like others PR.

File: src/main/java/com/khadri/jakartha/servlet/dao/ModifyDao.java
Comment (khadri2021) [inline src/main/java/com/khadri/jakartha/servlet/dao/ModifyDao.java:1]: packages should be com.khadri.stock.mobile.*;

**Pull request merged**